ATParser.cpp

```diff
@@ -155,7 +155,8 @@ int ATParser::vscanf(const char *format, va_list args)
         sscanf(_buffer+offset, _buffer, &count);
 
         // We only succeed if all characters in the response are matched
-        if (count == j) {
+        // and we are at the end of the response
+        if ((count == j) && (format[i-1] == (const char)c)) {
```

Comment:
In the past, we have been able to get around this by forcing sscanf to be greedy by matching the delimiter at the end but ignoring it. Something like this - https://github.com/sarahmarshy/mxchip-wifi-driver/blob/at-parser-port/mxchip/MXCHIP.cpp#L165